scanamo/src/main/scala/org/scanamo/ScanamoSync.scala

```diff
@@ -16,25 +16,16 @@
 
 package org.scanamo
 
-import cats.{ ~>, Id, Monad }
+import cats.{ ~>, Id }
 import org.scanamo.ops.*
 import software.amazon.awssdk.services.dynamodb.DynamoDbClient
 
 /** Provides a simplified interface for reading and writing case classes to DynamoDB
   *
   * To avoid blocking, use [[org.scanamo.ScanamoAsync]]
   */
-final class Scanamo private (client: DynamoDbClient) {
-  private val interpreter = new ScanamoSyncInterpreter(client)
-
-  /** Execute the operations built with [[org.scanamo.Table]]
-    */
-  def exec[A](op: ScanamoOps[A]): A = op.foldMap(interpreter)
-
-  /** Execute the operations built with [[org.scanamo.Table]] with effects in the monad `M` threaded in.
-    */
-  def execT[M[_]: Monad, A](hoist: Id ~> M)(op: ScanamoOpsT[M, A]): M[A] =
-    op.foldMap(interpreter andThen hoist)
+final class Scanamo private (client: DynamoDbClient) extends ScanamoClient(new ScanamoSyncInterpreter(client)) {
+  override def exec[A](op: ScanamoOps[A]): A = super.exec(op) // overridden to explicitly return `A` rather than `Id[A]`
```

Comment:
This override is necessary to ensure the _"Filters a table for attributes containing a specific substring"_ test (and presumably, other consumer code out in the wild) compiles under Scala 2.